analyzers/src/SonarAnalyzer.CSharp/Rules/DeclareTypesInNamespaces.cs

```diff
@@ -47,18 +47,15 @@ public sealed class DeclareTypesInNamespaces : DeclareTypesInNamespacesBase<Synt
             ((BaseTypeDeclarationSyntax)declaration).Identifier;
 
         protected override bool IsInnerTypeOrWithinNamespace(SyntaxNode declaration, SemanticModel semanticModel)
-        {
-            switch (declaration.Parent.Kind())
+            => declaration.Parent.Kind() switch
             {
-                case SyntaxKind.ClassDeclaration:
-                case SyntaxKind.StructDeclaration:
-                case SyntaxKind.NamespaceDeclaration:
-                case SyntaxKind.InterfaceDeclaration:
-                case SyntaxKindEx.RecordClassDeclaration:
-                    return true;
-                default:
-                    return false;
-            }
-        }
+                SyntaxKind.ClassDeclaration or
+                SyntaxKind.StructDeclaration or
+                SyntaxKind.NamespaceDeclaration or
+                SyntaxKind.InterfaceDeclaration or
+                SyntaxKindEx.RecordClassDeclaration or
+                SyntaxKindEx.FileScopedNamespaceDeclaration => true,
+                _ => false,
+            };
```

Comment:
It would be preferable to use `.IsAnyKind` in this case instead of the `switch` to remove syntax noise.

Reply:
Done.